### PR TITLE
Add hide 3-dot menu option

### PR DESCRIPTION
**Pull request by KTibow**
## Proposed change

Lets user hide 3 dot menu, mainly for kiosks.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue: #7614
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by bramkragten** [comment]
Please don't open a WIP PR when you are far from done. Open it if you want some eyes on it and have something that works.

**Comment by KTibow** [comment]
Sorry. It automatically generates the PR when I press the edit button.

**Comment by bramkragten** [comment]
You should not edit files with the edit button, definitely not PR's like this.

Besides that, this is not an option we will allow. We could consider a kiosk mode, but that should not work like this.

**Comment by KTibow** [comment]
Do you think? What should I use instead? The command line? I can tweak it to kiosk mode instead, if that's better.

File: translations/frontend/en.json [hunk 1]
```diff
@@ -3234,6 +3234,10 @@
                     "description": "This will hide the sidebar by default, similar to the mobile experience.",
                     "header": "Always hide the sidebar"
                 },
+                "kiosk_mode": {
+                    "description": "Hide the overflow menu (3 dots) and the sidebar button (you can still open it by pulling from the left).",
+                    "header": "Kiosk Mode"
+                },
```
Comment (zsarnett) [inline translations/frontend/en.json:3240]: Translations need to be done in `src\translations\en.json`

File: src/components/ha-button-menu.ts [hunk 2]
```diff
@@ -34,6 +37,7 @@ export class HaButtonMenu extends LitElement {
   }
 
   protected render(): TemplateResult {
+    this.style.display = this.hass.enableKioskMode ? "none" : "inline-block";
```
Comment (zsarnett) [inline src/components/ha-button-menu.ts:40]: Why is this being add to the Component for a case for an element that uses this component?

File: translations/frontend/en.json (hunk 1 above)
Comment (KTibow) [reply translations/frontend/en.json:3240]: Got it. Should I still keep them in the first place I put it, or should I try to revert the commit?

File: src/components/ha-menu-button.ts [hunk 3]
```diff
@@ -98,7 +98,7 @@ class HaMenuButton extends LitElement {
       return;
     }
 
-    this.style.display = newNarrow || this._alwaysVisible ? "initial" : "none";
+    this.style.display = (newNarrow || this._alwaysVisible) && !this.hass.enableKioskMode ? "initial" : "none";
```
Comment (zsarnett) [inline src/components/ha-menu-button.ts:101]: This should be done on the user of this component

Comment (KTibow) [reply src/components/ha-menu-button.ts:101]: Could you give me any ideas on how to do that? Still new to Polymer and this stuff.

**Comment by maykar** [comment]
`What should I use instead? The command line? `

@KTibow Ideally you'd be using a code editor like VS code with a Dev environment setup so that you can test your changes and also make large commits across files with meaningful commit messages.

https://developers.home-assistant.io/docs/frontend/development

**Comment by KTibow** [comment]
Closing because this PR wasn't made in a good way.

**Pull request closed without merging**